### ISSUE TITLE
Assertion failure for `debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js`

### ISSUE THREAD

**Issue by LOGQTAINIA** [v8-riscv/v8#152]
**Describe the bug**
Running `debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js` results in an assertion failure.

This bug shows up in #130  , but it could be triggered in simulated build now.

It is caused by nan-boxing PR.

**To Reproduce**
1. Build with options `riscv64.sim`
2. Run test/program 
```
/home/user/v8.snapshot-20200531/v8-cr/v8/out/riscv64.sim/d8 --test /home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js --random-seed=1706536607 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner --no-enable-one-shot-optimization --enable-inspector --allow-natives-syntax
```
3. See error

**Expected behavior**

**Errors/Logs**
```
Failure: expected <0> found <NaN>

Stack: MjsUnitAssertionError
    at assertEquals                      /home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js 405:7
    at success                           /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 23:38
    at DebugWrapper.listener             /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 110:11
    at DebugWrapper.invokeListener       /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 776:12
    at DebugWrapper.handleDebuggerPaused /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 745:10
    at DebugWrapper.dispatchMessage      /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 689:12
    at DebugWrapper.receiveMessage       /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 262:10 MjsUnitAssertionError
    at assertEquals                      /home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js 405:7
    at success                           /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 23:38
    at DebugWrapper.listener             /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 110:11
    at DebugWrapper.invokeListener       /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 776:12
    at DebugWrapper.handleDebuggerPaused /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 745:10
    at DebugWrapper.dispatchMessage      /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 689:12
    at DebugWrapper.receiveMessage       /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/test-api.js 262:10
/home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js:315: Failure: expected <null> found <MjsUnitAssertionError()>

Stack: MjsUnitAssertionError
    at assertNull /home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js 468:7
    at            /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 234:1
    throw new MjsUnitAssertionError(message);
    ^
MjsUnitAssertionError
    at assertNull /home/user/v8.snapshot-20200531/v8-cr/v8/test/mjsunit/mjsunit.js 468:7
    at            /home/user/v8.snapshot-20200531/v8-cr/v8/test/debugger/debug/side-effect/debug-evaluate-no-side-effect-builtins.js 234:1
```

**Additional context**

**Comment by penguinwu** [comment]
Fixed by #156

**Comment by penguinwu** [comment]
Fixed by #156
